File: Kernel/Logic/Logger.fs
namespace Kernel

(*
Logger:
    Helps us to observe what event happened inside our system.
    Global static object.

    Default output is through stdout.
    Using Godot output:
    ```
    Logger.setLogger GD.Print
    ```
*)
module Logger =
    open System

    let mutable private registeredLogger: Action<string> =
        new Action<string>(Console.WriteLine)

    let info s = registeredLogger.Invoke $"[INFO] {s}"

    let warn s = registeredLogger.Invoke $"[WARN] {s}"

    let setLogger logger = registeredLogger <- logger

File: mutemaanpa-cs/src/common/Logger.cs
using System;

namespace Mutemaanpa;

public static class Logger
{
    public static Action<string> endpoint { get; set; } = Console.WriteLine;

    public static void Info(string s)
    {
        endpoint($"[INFO] {s}");
    }

    public static void Warn(string s)
    {
        endpoint($"[WARN] {s}");
    }
}
